File: 프로그래머스/lv2/12981. 영어 끝말잇기/영어 끝말잇기.js
function solution(n, words) {
    let temp = {};
    
    temp[words[0]] = true;
    
    for(let i=1; i<words.length; i++){
        if(words[i-1].substr(-1) !== words[i][0] || temp[words[i]] ) {
            return [(i%n)+1 , Math.floor(i/n)+1]
         }
        temp[words[i]]= true;
    }
    
    return [0,0]
}

File: 프로그래머스/2/12981. 영어 끝말잇기/영어 끝말잇기.java
import java.util.HashMap;

class Solution {
    public int[] solution(int n, String[] words) {
        HashMap<String, Boolean> map = new HashMap<>();

        int index = 0;

        map.put(words[0], true);

        for (int i = 1; i < words.length; i++) {
            String pre = words[i - 1];
            String cur = words[i];
            char first = cur.charAt(0);
            char last = pre.charAt(pre.length() - 1);

            if (map.containsKey(cur) ||  last != first) {
                index = i;
                break;
            } else {
                map.put(cur, true);
            }
        }
        
        if (index == 0) {
            return new int[]{0,0};
        }
        return new int[]{(index)%n + 1,(int)Math.ceil(((double)index+1)/n)};
    }
}